incubator/binding-mqtt/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -2171,9 +2214,10 @@ private void doSendSessionState(
             newState.forEach(subscription ->
                 sessionStateBuilder.subscriptionsItem(subscriptionBuilder ->
                 {
-                    subscriptionBuilder.pattern(subscription.filter);
                     subscriptionBuilder.subscriptionId(subscription.id);
+                    subscriptionBuilder.qos(subscription.qos);
                     subscriptionBuilder.flags(subscription.flags);
+                    subscriptionBuilder.pattern(subscription.filter);
```

Comment:
Rename `subscriptionBuilder` to `s` (lambda variable) and use fluent style to set `pattern`, `subscriptionId`, etc in a single statement.